app/scripts/modules/core/src/pipeline/config/stages/deploy/deployStage.js

```diff
@@ -33,6 +33,7 @@ module.exports = angular.module('spinnaker.core.pipeline.stage.deployStage', [
             }
             return stage.clusters.every(cluster =>
               cloudProviderRegistryProvider.$get().getValue(cluster.provider, 'serverGroup.skipUpstreamStageCheck')
+              || cluster.imageSource === 'artifact'
```

Comment:
This feels a little tacked on - might be better to include a flag in the the provider's config (like `skipUpstreamStageCheck`).